File: get_hardware.cpp

#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream cpuinfo("/proc/cpuinfo");
    std::string line;
    std::cout << "CPU Info:\n";
    while (std::getline(cpuinfo, line)) {
        if (line.find("model name") != std::string::npos) {
            std::cout << line << std::endl;
        }
    }

    std::ifstream meminfo("/proc/meminfo");
    std::cout << "\nMemory Info:\n";
    while (std::getline(meminfo, line)) {
        if (line.find("MemTotal") != std::string::npos ||
            line.find("MemAvailable") != std::string::npos)
            std::cout << line << std::endl;
    }

    return 0;
}

File: get_hardware.rs

use std::fs;

fn main() {
    let cpu = fs::read_to_string("/proc/cpuinfo").unwrap();
    for line in cpu.lines() {
        if line.contains("model name") {
            println!("{}", line);
        }
    }

    let mem = fs::read_to_string("/proc/meminfo").unwrap();
    for line in mem.lines() {
        if line.contains("MemTotal") || line.contains("MemAvailable") {
            println!("{}", line);
        }
    }
}
